### PR TITLE
Formatter suggestion

### LINKED ISSUE #525: Code Formatting Guide Needed
As seen in https://github.com/signum-network/signum-node/pull/469#discussion_r594667678 there exists a need for a formatting guideline or requirements document. I suggest basing the formatting on the most popular (or official, if any exists) java formatting standards and creating a FORMATTING.md or section of CONTRIBUTING.md that explains how formatting should be accomplished in this project.

Also, many IDEs have the ability to read auto-formatting rules from a file included in the project directory. Consider including formatting rules files for the most popular IDEs that support this option.

Checking for proper formatting should be done with an automated linter, if possible, that runs in a github workflow on every PR submission and update to ensure formatting is correct.

Here is a good example of a github action that can be deployed to automatically format the code into Google Java Format https://github.com/marketplace/actions/google-java-format. Not saying that's what we should use, but it's a good example. This completely automates the formatting and doesn't care what developers do in their own code. The --dry-run option is interesting as it does not actually commit the formatting changes. It could possibly be used to simply check for proper formatting and then fail the PR's workflow run if the formatting isn't correct.

### PR DESCRIPTION
**Pull request by damccull**
This is a proof of concept of using a formatter and style guide to force formatting a specific way.

In this PR, I have added two new files:
1. `eclipse-java-style.xml` - contains auto-formatting rules based on Google java style.
    This file is an Eclipse formatter xml, which many IDEs can use, eclipse and vscode included.
3. `checkstyle.xml` - contains style lints for the CheckStyle program/extension based on Google java style.

I have also formatted Burst.java in accordance with the rules. I think it makes things more readable and explicit.

My suggestion is to adopt the very slightly modified Google java style and format all source code to this new style. The work of reformatting can be done over time, but new code should all be written to the standard.

The CheckStyle program is capable of being used in CI runners and GH actions to force a style check and fail a PR/commit if the format is not correct. I have not enabled this in this proof of concept PR. If this policy is adopted, reformatting of the code base in its entirety will need to be completed before using CI to check, or every build will fail every time.

Future work if approved:
1. Update contributing guide to explain that code must be correctly formatted, and how this can be done in several different editors using the provided files.
3. Reformat the entire code base.
4. Enable CheckStyle in GH with style errors failing the build.

Please look this over and let me know what you think. I tried to make each change type in the PR a separate commit for ease of reading the differences, instead of having to read the entire file changes at once.

Closes #525 if approved.

This is mutually exclusive with #788  at the moment, but I will rebase whichever doesn't get approved first and push corrections.

### CODE REVIEW

**Comment by ohager** [comment]
I merged the other PR before this one. I think linting (wording in JS world) make sense...

**Comment by damccull** [comment]
I have rebased this on main, and I've included a devcontainer and .vscode workspace settings that can be used with vscode and docker, or with github codespaces as-is. They are set to automatically add the required extensions and settings for vscode to be a full dev environment for the node, with appropriate styling and style-linting. I can add instructions in the future or, if desired, remove these two files from the PR.

Either way, I recommend we consider providing sane defaults for workspaces in vscode, eclipse, and perhaps one more popular IDE.

I'll also add a checkstyle stage to the GH actions after full formatting of the repo is complete, and assuming this gets accepted.

**Comment by ohager** [comment]
I think it's a good practice to have linters... we might think of using git hooks to apply auto-linting on commits. Never done in context of Java (but I use husky hooks on NodeJS world heavily)

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (ohager) [inline .vscode/settings.json:1]: usually, it's not recommended to add IDE specific setting files. I'm kind of opposed to add this here.

**Review by ohager — CHANGES_REQUESTED** [review]
I don't think vscode folder should be added here.

File: .vscode/settings.json (hunk 1 above)
Comment (damccull) [reply .vscode/settings.json:1]: I'll yank that file for now, but I'll add here my comments from discord:

I normally agree, but these are workspace settings. They shouldn't interfere with any user settings. We can make it part of the contribution to ask people not to change that file.

Vs code has 3 levels of settings, including a user settings so it shouldn't be a big deal for people to customize them.
I personally think .vscode folder should be committed, and instructions on why it's there added to most projects. It's a great way to share project specific settings that need to exist.
In this case, we can nuke it in favor of instructions on how to set it up and use it yourself, but I'm 100% sure use cases like this are why vscode has layers of settings. Doesn't idea?

Comment (damccull) [reply .vscode/settings.json:1]: In this case, without the .vscode folder, devs will have to manually set up the formatter and checkstyle settings with each new clone. We can include instructions for that, but I find it a better choice to give them the settings by default in the 'Workspace' settings, and instruct them to use their 'User' or 'System' settings for personal customizations.

Comment (damccull) [reply .vscode/settings.json:1]: I'll make it a separate PR to be discussed on, though.

Comment (damccull) [reply .vscode/settings.json]: I've removed the devcontainer and vscode settings folders and will send separate PR for them.

**Review by ohager — APPROVED** [review]
gj

**Pull request merged**